### DISCUSSION TITLE
`skia_safe::textlayout::TextStyle::font_metrics` brokes in 0.70.0

### DISCUSSION THREAD

**Discussion by FlysoftBeta** [rust-skia/rust-skia#934] [Q&A] [answered]
in 0.70.0, the result of `font_metrics` is an empty struct, but in 0.69.0, it works fine.

```rust
let mut style = skia_safe::textlayout::TextStyle::new();
style.set_foreground_paint(skia_safe::Paint::new(skia_safe::Color4f::new(1.0, 1.0, 1.0, 1.0), skia_safe::ColorSpace::new_srgb()));
style.set_font_size(20.0);
style.set_font_style(skia_safe::FontStyle::new(
    500.0.into(),
    skia_safe::font_style::Width::NORMAL,
    skia_safe::font_style::Slant::Upright,
));
dbg!(style.clone(), style.font_metrics());
```

**Comment by pragmatrix** [comment, marked as answer]
There were some changes with the default font / typeface (empty since 0.70.0) which I suspect is the underlying cause. I suggest that you use `TextStyle::set_typeface()`. And to get a the default typeface use `FontMgr::new()` and `FontMgr::legacy_make_typeface()`. The strategy here is to be explicit on what font to use, best to deploy one with your app, or query one using the `FontMgr::match...` functions.

**Reply by FlysoftBeta** [reply]
Sorry for the delayed response. Setting a typeface does the job. Thanks!
